### PR TITLE
CMakeLists.txt: debug PG refs in debug builds

### PR DESCRIPTION
**Pull request by liewegas**
I can't decide if this is a good idea.  On one hand, it will help us find leaks the first time they happen in qa.  On the other hand, this moves the debug build further away from being usable in production.  And I /thank/ i squashed the pg leak...

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -345,6 +345,7 @@ if(WITH_LZ4)
 endif(WITH_LZ4)
 
 option(WITH_CEPH_DEBUG_MUTEX "Use debug ceph::mutex with lockdep" OFF)
+option(WITH_DEBUG_PG_REFS "Debug PG references" OFF)
```
Comment (tchaikov) [inline CMakeLists.txt]: we already have an option for this

https://github.com/ceph/ceph/blob/c1112928b7926fdc0a53d254a5f72055728f3b58/CMakeLists.txt#L572

which sets 

https://github.com/ceph/ceph/blob/c1112928b7926fdc0a53d254a5f72055728f3b58/src/include/config-h.in.cmake#L280-L281

**Comment by liewegas** [comment]
The problem I've had is that if you do see it you have to push a special build/branch that enables the debugging, and then figure out how to reproduce it.  If it's on all the time, then you already have the log from that one failure to track it down.  FWIW i recently turned set debug_refs=5 for the valgrind runs so that when a leak is found we'll have a chance of finding it (if it's a RefCountedObject that enables ref debug output).

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
retest this please

**Comment by neha-ojha** [comment]
retest this please

**Comment by jdurgin** [comment]
retest this please

**Comment by jdurgin** [comment]
retest this please

**Pull request closed without merging**